### PR TITLE
Address C compiler warnings

### PR DESCRIPTION
**Pull request by nickdrozd**
This is mostly adding missing prototypes, fixing https://github.com/Wilfred/remacs/issues/957.

### CODE REVIEW

File: src/buffer.c [hunk 1]
```diff
@@ -4973,20 +4976,6 @@ defvar_per_buffer (struct Lisp_Buffer_Objfwd *bo_fwd, const char *namestring,
...
 
-/* Similar to defvar_lisp but define a variable whose value is the
-   Lisp_Object stored in the current buffer.  LNAME is the Lisp-level
-   variable name.  VNAME is the name of the buffer slot.  PREDICATE
-   is nil for a general Lisp variable.  If PREDICATE is non-nil, then
-   only Lisp values that satisfies the PREDICATE are allowed (except
-   that nil is allowed too).  DOC is a dummy where you write the doc
-   string as a comment.  */
-
-#define DEFVAR_PER_BUFFER(lname, vname, predicate, doc)		\
```
[the remark this answers is not in the document]
Comment (nickdrozd) [reply src/buffer.c]: No, it was defined twice. I think it was just copied by mistake.

File: configure.ac [hunk 2]
```diff
@@ -908,7 +908,7 @@ AC_CACHE_CHECK([whether the compiler is clang], [emacs_cv_clang],
      [emacs_cv_clang=yes],
      [emacs_cv_clang=no])])
 
-WERROR_CFLAGS=
+WERROR_CFLAGS=-Werror
```
[the remark this answers is not in the document]
Comment (shaleh) [reply configure.ac]: Agreed

[the remark this answers is not in the document]
Comment (shaleh) [reply configure.ac]: We could add a 'make strict' or equivalent to make it easier for people to compile like Travis will. Travis could then use 'make strict'.

**Comment by nickdrozd** [comment]
@Wilfred @shaleh I updated it to only run `-Werror` on Travis. I think it works? The makefile works locally anyway.

**Comment by nitishch** [comment]
#957 can be closed now.

**Pull request merged**